### PR TITLE
Register needed extension by default jira#709

### PR DESCRIPTION
**Pull request by MalloZup**
# description:

see jira.

### CODE REVIEW

**Comment by MalloZup** [comment]
tested on sle15 works. I will test just for sanity on sle12

File: salt/os_setup/registration.sls [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {% if grains['os_family'] == 'Suse' %}
 {% if not grains.get('qa_mode') or '_node' not in grains.get('role') %}
 {% if grains['reg_code'] %}
+{% set mod_reg_code = grains['reg_code'] %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:4]: Why do you call it `mod_reg_code` to this? Call it `reg_code` in any case. And if you do this, replace all the following `grains['reg_code']` by it

File: salt/os_setup/registration.sls [hunk 2]
```diff
@@ -11,6 +13,57 @@ register_system:
         interval: 15
 {% endif %}
 
+
+{% if not 'iscsi_srv' in grains.get('role') %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:17]: Why are you excluding the `iscsi_srv` here? And only for SLE12?
I would let the `iscsi` server to be registered, I don't see anything wrong with that

File: salt/os_setup/registration.sls [hunk 3]
```diff
@@ -11,6 +13,57 @@ register_system:
...
+{% if '12' in grains['osrelease'] %}
+{% if grains['osrelease'] == '12' %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:19]: Don't use `osrelease`. Use `osrelease_info` or `osmajorrelease` better. It returns a list with OS release numbers. Example
`[12, 4]` or `[15]`

File: salt/os_setup/registration.sls [hunk 4]
```diff
@@ -11,6 +13,57 @@ register_system:
...
+    - retry:
+        attempts: 3
+        interval: 15
+
+{% endif %}
+{% endif %}
+
+# sle15 
+# temporary PackageHub (temporarily for GCP until ECO-1148 is released)
+{% if '15' in grains['osrelease'] %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:53]: The same than for SLE12, don't use `osrelease`.
In fact, we don't need to duplicate this `PackageHub` code. If it's needed by SLE12 and SLE15, and both use the same code, just use a unique state, the `osrelease` and `arch` will do the work for that.

File: salt/os_setup/registration.sls [hunk 5]
```diff
@@ -11,6 +13,57 @@ register_system:
...
+  cmd.run:
+    - name: /usr/bin/SUSEConnect -p sle-module-public-cloud/{{ grains['osrelease'] }}/{{ arch }} {{ "-r $mod_reg_code" if mod_reg_code else "" }}
+    - env:
+        - mod_reg_code: {{ mod_reg_code }}
+    - retry:
+        attempts: 3
+        interval: 15
+
+# temporary PackageHub (temporarily for GCP until ECO-1148 is released)
+default_PackageHub_registration_sle12:
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:39]: Why are you adding this `sle12` sufix to the state name? I think it's not needed and breaks the naming used in the other registration names

File: salt/os_setup/registration.sls [hunk 6]
```diff
@@ -1,16 +1,51 @@
...
     - env:
-        - reg_code: {{ grains['reg_code'] }}
+        - reg_code: {{ reg_code }}
     - retry:
         attempts: 3
         interval: 15
 {% endif %}
 
+
+{% if 12 == grains['osmajorrelease'] %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:17]: This order is a bit strange. Usually I would put `{% if grains['osmajorrelease'] == 12 %}`

**Comment by MalloZup** [comment]
@arbulu89 let me know thx!

File: salt/os_setup/registration.sls [hunk 7]
```diff
@@ -1,16 +1,67 @@
...
-        - reg_code: {{ grains['reg_code'] }}
+        - reg_code: {{ reg_code }}
     - retry:
         attempts: 3
         interval: 15
 {% endif %}
 
+
+{% if grains['osmajorrelease'] == 12 %}
+# hardcode the 12 version number for the 2 follo
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:18]: I think that the comment is unfinished

File: salt/os_setup/registration.sls [hunk 8]
```diff
@@ -1,16 +1,67 @@
...
+    - env:
+        - reg_code: {{ reg_code }}
+    - retry:
+        attempts: 3
+        interval: 15
+
+{% endif %}
+{% endif %}
+
+{% if grains['osmajorrelease'] == 15 %}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:41]: I think we can change the `if` by an `elif` combining with the previous `if`. It cannot be anything else than SLE12 or SLE15. 
Besides, we can have the provider condition in the same line too. Something like:
```
{% elif grains['osmajorrelease'] == 15 and grains['provider'] in ['gcp', 'aws', 'azure'] %}
```
If you do this, don't forget to remove the previous `endif`

File: salt/os_setup/registration.sls (hunk 7 above)
Comment (MalloZup) [reply salt/os_setup/registration.sls:18]: ah yes forgot it

File: salt/os_setup/registration.sls [hunk 9]
```diff
@@ -1,16 +1,64 @@
...
         attempts: 3
         interval: 15
 {% endif %}
 
+
+{% if grains['osmajorrelease'] == 12 %}
+# hardcode the 12 version number for the 2 following modules, since they don't offer a sp version only 1. 
+default_sle_module_adv_systems_management_registration:
+  cmd.run:
+    - name: /usr/bin/SUSEConnect -p sle-module-adv-systems-management/12/{{ arch }} {{ "-r $reg_code" if reg_code else "" }}
```
Comment (arbulu89) [inline salt/os_setup/registration.sls:21]: By the way @MalloZup , we can replace `{{ "-r $reg_code" if reg_code else "" }}` by `-r $reg_code` simply. If we are in this piece of code the `reg_code` always exists.

Comment (MalloZup) [reply salt/os_setup/registration.sls:21]: yes indeed thx!

Comment (MalloZup) [reply salt/os_setup/registration.sls]: done

**Pull request merged**